pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -202,6 +209,8 @@ private void loadCreationMeta(File crcFile) throws IOException {
   }
 
   private void init() {
+    _creatorName = _segmentMetadataPropertiesConfiguration.getString(Segment.SEGMENT_CREATOR_VERSION);
```

Comment:
This might throw exception if the property is not present in the metadata.properties. You might want to either check if the configuration contains this property or pass in default null value.